File: GeeksForGeeks/School/Equal_to_product.cpp
#include <iostream>
#include <set>
using namespace std;

int main() {
	int test_case;
    cin >> test_case;
    while (test_case--) {
        // Reading the inputs
        int size, product;
        cin >> size >> product;
        int arr[size], i;
        for (i = 0; i < size; i++) {
            cin >> arr[i];
        }

        // Logic Part
        set<int> container;
        bool result_flag = false;
        for (i = 0; i < size; i++) {
            if (arr[i] == 0) {
                continue;
            }
            auto search_result = container.find(product / arr[i]);
            if ((search_result != container.end()) &&
                    (*search_result * arr[i] == product)) {
                result_flag = true;
                break;
            }
            container.insert(arr[i]);
        }
        if (result_flag) {
            cout << "Yes" << endl;
        }
        else {
            cout << "No" << endl;
        }
    }
	return 0;
}

File: GeeksForGeeks/School/Equal_to_product.java
import java.util.HashSet;
import java.util.Scanner;
import java.util.Set;

class GFG
 {
	public static void main (String[] args)
	{
	    Scanner scan = new Scanner(System.in);
	    int testcase = scan.nextInt();
    	while (testcase != 0)
    	{
            int size, product;
            size = scan.nextInt();
            product = scan.nextInt();

            int[] arr = new int[size];
            int i;
            for (i = 0; i < size; i++)
            {
                arr[i] = scan.nextInt();
            }

            Set<Integer> container = new HashSet<>();
            boolean resultFlag = false;
            for (i = 0; i < size; i++)
            {
                if (arr[i] == 0) { continue; }
                int toBeSearched = product / arr[i];
                if ((toBeSearched * arr[i] == product) && container.contains(toBeSearched))
                {
                    resultFlag = true;
                    break;
                }
                container.add(arr[i]);
            }
            if (resultFlag) {
                System.out.println("Yes");
            }
            else {
                System.out.println("No");
            }
            --testcase;
    	}
    	scan.close();
	}
}